### DISCUSSION TITLE
How to change background color and other theming options?

### DISCUSSION THREAD

**Discussion by bastianolea** [surveydown-dev/surveydown#139] [Q&A]
### Description

I'm trying to change the background color of surveys, along with other colors and styles. Changing `backgroundcolor` in `survey.qmd` does nothing, neither does `barcolor` 🙁

**Comment by jhelvy** [comment]
You can modify most of this in the yaml section at the top of your `survey.qmd` file.

There is no yaml item for background color. To do that you'll need to supply a custom css file, like [this](https://surveydown.org/survey-components#changing-the-look-and-feel).

Changing the bar color should be doable like [this](https://surveydown.org/survey-components#progress-bar). (link edited)

**Comment by bastianolea** [comment]
Thanks for the quick answer!

Using `theme: [united, custom.scss]` gives me `Error in if (theme == "default") { : the condition has length > 1`, so I can only use a theme _or_ a custom .scss.

If I only use custom.scss as theme, I am able to change some aspects of the style, like `$primary` and `$body-color`, but I can't seem to change the background color. 

PS: one of your links seems to be a mostly empty repository.

**Comment by jhelvy** [comment]
Oh good catch, I think that is indeed a bug on the theme thing. @pingfan-hu do you want to take a look at this? Probably just a simple bug checking the theme, plus I'm not sure why a background color isn't available in a custom SCSS file.

And can you tell me which link seems to be a mostly empty repo?

**Reply by bastianolea** [reply]
It was the link for the bar color (https://github.com/jhelvy/emse-student-faq), but then I realized changing bar color using `barcolor` does indeed work, my mistake, I was expecting the empty bar color to change as well (that brings me the question of how to change the empty bar color, which seems to be `#ECE8DF`)

**Reply by jhelvy** [reply]
Haha woops, that was a copy-paste error. Completely unrelated link. I meant to link to [here](https://surveydown.org/survey-components#progress-bar). That section shows how to change the bar color. We decided to include things that are specific to surveydown in the yaml (`barposition` and `barcolor`), and otherwise the rest of the yaml should work just like a regular Quarto doc (e.g. how to change the themes). Obviously we have a bug there though on providing a custom theme.

**Comment by jhelvy** [comment]
Just noting this is still on our radar. We have a few other larger mods going on so we'll get to this when those are done. Just merged PR #134 to support multiple languages in the system messages, and we're also working on restoring the session when you refresh the page.

**Comment by pingfan-hu** [comment]
Hi @bastianolea ! Sorry for the late reply. I was trapped with the cookies thing.
Now the theme thing has been solved. You are free to define a list of themes just like a normal `qmd` file.
You can test it with `theme: [united, custom.scss]` in YAML, or any other combinations of `bootswatch` themes and `scss` files.

Please install our latest package and test it out:
```r
remotes::install_github("surveydown-dev/surveydown", force = TRUE)
```

We'll soon publish this update to CRAN as well.

**Comment by jhelvy** [comment]
This is handled in the [yaml options](https://surveydown.org/manuals/survey-components#changing-the-look-and-feel
) in the survey.qmd file.
